crates/wasi-common/src/hostcalls_impl/misc.rs

```diff
@@ -218,6 +218,10 @@ pub(crate) fn poll_oneoff(
 
     let mut timeout: Option<ClockEventData> = None;
     let mut fd_events = Vec::new();
+
+    if subscriptions.is_empty() {
```

Comment:
Could you reference the PR this was introduced in in WebAssembly/WASI crate? I reckon this might come in handy in the future should we ever need to track the API against the spec changes :-)

Reply:
In my opinion mentioning the spec changes in the code will generate too much noise if done in a consequent manner, but I cited the spec. Let's see what others say

Reply:
That sounds fair. I'm happy either way then if you feel that it will generate too much noise :-)

Reply:
I agree with you both :-). We don't need to cite the spec on every detail, but it is nice to do for particularly surprising details, and I think this one qualifies.

Reply:
I'm happy with the current comment.  If everyone else is, I think we can resolve this and merge.